packages/common/src/content/contentUtils.ts

```diff
@@ -437,3 +440,52 @@ const getContentRelativeUrl = (
     content.typeKeywords
   );
 };
+
+// Tests can be found in packages/common/test/content/content.test.ts
+/**
+ * Get the status of a content item
+ * @param entity the content item
+ * @returns the status of the content item
+ */
+export async function getContentStatus(
```

Comment:
this really feels to me like it should be `getServiceStatus(url: string, requestOptions: IRequestOptions)`
 
it also feels like it should optionally allow users to override the default timeout.

So probably the best thing to do is extend [IGetLayerOptions](https://esri.github.io/arcgis-rest-js/api/feature-layer/IGetLayerOptions/) like this:

```ts
interface IGetServiceStatusOptions extends IGetLayerOptions {
  /* number of milliseconds to wait for the service response, defaults to 3000 */
  timeout?: number
}

export async function getServiceStatus(options: IGetServiceStatusOptions) {
  const { timeout, ...requestOptions } = options;
  // ....
}
````